src/main/kotlin/graphics/scenery/Settings.kt

```diff
@@ -9,30 +13,93 @@ import java.util.concurrent.ConcurrentHashMap
  *
  * @author Ulrik Günther <hello@ulrik.is>
  */
-class Settings(override var hub: Hub? = null) : Hubable {
+class Settings(override var hub: Hub? = null, prefix : String = "scenery.", propertiesFilePath : String? = null) : Hubable {
     private var settingsStore = ConcurrentHashMap<String, Any>()
     private val logger by LazyLogger()
 
+    var settingsUpdateRoutines : HashMap<String, ArrayList<() -> Unit>> = HashMap()
+    lateinit var propertiesFile : File
```

Comment:
I think it might make sense not to give a path as `String` to the class, but instead an InputStream. This way, the properties could be read from whereever, JAR file, URL, etc. That would also remove the necessity for the lateinit var here.

Reply:
I c. I wasnt aware that a InpoutStream is a hashable object